File: digitalmaktabapi/Models/Course.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace digitalmaktabapi.Models
{
    [Table("Course")]
    public class Course : Base
    {
        public required Guid ClassId { get; set; }
        public required Class Class { get; set; }
        public required Guid SubjectId { get; set; }
        public required Subject Subject { get; set; }
    }
}

File: digitalmaktabspa/src/models/Course.ts
import { CourseSection } from "./CourseSection";
import { Base } from "./Base";
import { Class } from "./Class";
import { Subject } from "./Subject";
import { Teacher } from "./Teacher";

export interface Course extends Base {
  classId: string;
  class: Class;
  subjectId: string;
  subject: Subject;
  teacherId: string;
  teacher: Teacher;
  courseSections: CourseSection[];
}
